sklearn/utils/estimator_checks.py

```diff
@@ -454,8 +455,37 @@ def check_dict_unchanged(name, Estimator):
                               'Estimator changes __dict__ during %s' % method)
 
 
+def check_fit_adds_private_attributes_only(name, Estimator):
+    rnd = np.random.RandomState(0)
+    X = 3 * rnd.uniform(size=(20, 3))
+    y = X[:, 0].astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    estimator = Estimator()
+    set_testing_parameters(estimator)
+
+    if hasattr(estimator, "n_components"):
+        estimator.n_components = 1
+    if hasattr(estimator, "n_clusters"):
+        estimator.n_clusters = 1
+
+    set_random_state(estimator, 1)
```

Comment:
this part seems to be very repetitive, so maybe it's possible to refactor / extract method here.. not sure though